iothub/device/src/Authentication/IotHubConnectionCredentials.cs

```diff
@@ -392,28 +385,34 @@ private void Validate()
                     }
                 }
             }
+
+            if (SasTokenTimeToLive > TimeSpan.FromHours(1))
+            {
+                if (Logging.IsEnabled)
+                    Logging.Info(this, $"You are using a larger value {SasTokenTimeToLive} than our recommendation 1 hour for the SAS token TTL.");
+            }
+
+            if (SasTokenRenewalBuffer > 85)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Info(this, $"You are using a larger percent of TTL {SasTokenRenewalBuffer} as the token renewal buffer. " +
+                        "Your token will be renewed more frequently.");
+            }
         }
 
         private static IAuthenticationMethod GetAuthenticationMethodFromConnectionString(IotHubConnectionString iotHubConnectionString)
         {
-            if (iotHubConnectionString.SharedAccessKeyName != null)
```

Comment:
with the new `ClientAuthenticationWithSakRefresh`, SharedAccessKeyName is optional so nullable.